server/automute_preferences.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/plugin"
+)
+
+func (p *Plugin) PreferencesHaveChanged(c *plugin.Context, preferences []model.Preference) {
+	p.updateAutomutingOnPreferencesChanged(c, preferences)
+}
+
+func (p *Plugin) updateAutomutingOnPreferencesChanged(c *plugin.Context, preferences []model.Preference) {
+	userIDsToEnable, userIDsToDisable := getUsersWhoChangedPlatform(preferences)
+
+	for _, userID := range userIDsToEnable {
+		if connected, err := p.isUserConnected(userID); err != nil {
+			p.API.LogError(
+				"Unable to potentially enable automute for user",
+				"UserID", userID,
+				"Error", err.Error(),
+			)
+		} else if !connected {
+			continue
+		}
+
+		if _, err := p.enableAutomute(userID); err != nil {
+			p.API.LogError(
+				"Unable to mute channels for a user who set their primary platform to Teams",
+				"UserID", userID,
+				"Error", err.Error(),
+			)
+		}
+	}
+
+	for _, userID := range userIDsToDisable {
+		_, err := p.disableAutomute(userID)
+		if err != nil {
+			p.API.LogError(
+				"Unable to unmute channels for a user who set their primary platform to Mattermost",
+				"UserID", userID,
```

Comment:
```suggestion
				"user_id", userID,
```

Reply:
I'll go through and do a pass on these since this style will come up in all 3 of my PRs, but I saw examples of both naming conventions (and I swear there was one rogue `userId`), so it would be good to get the rest of those consistent at some point

Reply:
See https://github.com/mattermost/mattermost-plugin-msteams-sync/pull/480, but this should have mostly fixed everything (done after your branch was started, of course!)

Reply:
Ah, that probably explains why it was so much harder for me to find an example that matched mine when I searched recently